pkg/controller/installation/defaults_test.go

```diff
@@ -121,6 +129,36 @@ var _ = Describe("Defaulting logic tests", func() {
 		instanceCopy := instance.DeepCopyObject().(*operator.Installation)
 		fillDefaults(instanceCopy)
 		Expect(instanceCopy.Spec).To(Equal(instance.Spec))
+		Expect(validateCustomResource(instance)).NotTo(HaveOccurred())
+	})
+
+	It("should default BGP to enabled for Calico CNI", func() {
+		instance := &operator.Installation{
+			Spec: operator.InstallationSpec{
+				CNI: &operator.CNISpec{
+					Type: operator.PluginCalico,
+				},
+			},
+		}
+		fillDefaults(instance)
+		Expect(*instance.Spec.CalicoNetwork.BGP).To(Equal(operator.BGPEnabled))
+		Expect(validateCustomResource(instance)).NotTo(HaveOccurred())
+	})
+
+	It("should default BGP to disabled for other CNI plugins", func() {
+		instance := &operator.Installation{
+			Spec: operator.InstallationSpec{
+				CNI: &operator.CNISpec{
+					Type: operator.PluginAmazonVPC,
+				},
+				CalicoNetwork: &operator.CalicoNetworkSpec{},
+			},
+		}
+		fillDefaults(instance)
+		Expect(*instance.Spec.CalicoNetwork.BGP).To(Equal(operator.BGPDisabled))
```

Comment:
I don't think fillDefaults should be creating an invalid configuration. This seems like a problem.

Reply:
I see that the initial instance is invalid currently too, so I guess that's fine. I think we should include the Expect below and just have it expecting an error for now. Then when the validation is changed this test will fail and get updated.

Reply:
Yeah, exactly. `fillDefaults` is fine, it's the initial configuration that we're passing in that's invalid. 

I'll fix the assertion.